fsspec/config.py

```diff
@@ -21,12 +26,24 @@ def set_conf_env(conf_dict, envdict=os.environ):
     envdict : dict-like(str, str)
         Source for the values - usually the real environment
     """
+    kwarg_keys = []
     for key in envdict:
-        if key.startswith("FSSPEC"):
-            if key.count("_") < 2:
+        if key.startswith("FSSPEC_") and len(key) > 7 and key[7] != "_":
+            if key.count("_") > 1:
+                kwarg_keys.append(key)
                 continue
-            _, proto, kwarg = key.split("_", 2)
-            conf_dict.setdefault(proto.lower(), {})[kwarg.lower()] = envdict[key]
+            try:
+                value = json.loads(envdict[key])
+            except json.decoder.JSONDecodeError:
+                pass
```

Comment:
Should we log this exception?  Silently ignoring the error seems like it could be annoying, especially given how strict the JSON parser is.

Reply:
Yes, this should be logged. A logger can exist specifically in this module, as there are a number of things that we may want to output about the config. A would even consider upping this to a warning, since the possibility of using this method is new, and we want to make people use valid configs.

Reply:
It is now a warning.